sql/core/src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java

```diff
@@ -291,7 +291,7 @@ public void testSetOperation() {
       unioned.collectAsList());
 
     Dataset<String> subtracted = ds.except(ds2);
-    Assert.assertEquals(Arrays.asList("abc", "abc"), subtracted.collectAsList());
+    Assert.assertEquals(Arrays.asList("abc"), subtracted.collectAsList());
```

Comment:
This is pretty strange. I will check the exact behavior of our current `Except` operator. It sounds like it does not remove the duplicate.